### PR TITLE
Add Zoro.to presence

### LINKED ISSUE #3976: Zoro | zoro.to
**Prerequisites and essential questions** 
- [x] Is it a popular site? 
- [x] Is the website older than 2 months?
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
azu#3000

**What's the name of the service?** https://zoro.to

**What should the Presence display?** A "watch on zoro.to" button in which if you click it will redirect you to the episode you are watching on the site

**If possible, please provide a logo for the service (512x512 minimum)**

### PR DESCRIPTION
**Pull request by vahya**
Closes PreMiD/Presences#3976

Screenshot 2021-06-30 024519
Screenshot 2021-06-30 024541
Screenshot 2021-06-30 024603
Screenshot 2021-06-30 024641
Screenshot 2021-06-30 024705
Screenshot 2021-06-30 024735

### CODE REVIEW

File: websites/Z/Zoro/iframe.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
+  const video: HTMLVideoElement = document.querySelector("video");
+  if (video && !isNaN(video.duration)) {
+    iframe.send({
+      iFrameVideo: true,
```
Comment (ririxidev) [inline websites/Z/Zoro/iframe.ts:7]: for what purpose, you need this variable?

File: websites/Z/Zoro/presence.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          url: document.location.href
+        }
+      ];
+      break;
+    }
+    case "/events": {
+      presenceData.details = "Looking at events";
+      break;
+    }
+    case pathname.startsWith("/event/") && pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:145]: this case statement doesn't have any sense. You are checking if pathname starts with "/event" so basically it is already a thing?
Do
```suggestion
    case pathname?.startsWith("/event/"): {
```

File: websites/Z/Zoro/presence.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        presenceData.buttons = [
+          {
+            label: "Join Room",
+            url: document.location.href
+          }
+        ];
+      }
+      break;
+    }
+    case pathname.startsWith("/watch") && pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:116]:
```suggestion
    case pathname?.startsWith("/watch"): {
```

File: websites/Z/Zoro/presence.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+      if (pathname !== "/az-list") {
+        presenceData.state = `Titles starting with ${
+          pathname.substring(9) === "other"
+            ? "Other characters"
+            : `${pathname.substring(9)}`
+        }`;
+      }
+      break;
+    }
+    case pathname.startsWith("/watch2gether/") && pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:91]:
```suggestion
    case pathname.startsWith("/watch2gether/"): {
```

File: websites/Z/Zoro/presence.ts [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+      const profile: HTMLDivElement = document.querySelector("div.ph-title"),
+        link: HTMLAnchorElement = document
+          .querySelector("ul.nav.nav-tabs.pre-tabs")
+          .querySelector("a.nav-link.active");
+      if (profile) presenceData.details = `Viewing User: ${profile.innerText}`;
+      if (link) presenceData.state = `At ${link.innerText}`;
+      break;
+    }
+    case (pathname.startsWith("people") || pathname.startsWith("character")) &&
+      pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:70]:
```suggestion
    case (pathname?.startsWith("people") || pathname?.startsWith("character")): {
```

File: websites/Z/Zoro/presence.ts [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+      }
+      break;
+    }
+    case "/search": {
+      presenceData.details = "Searching";
+      const { search } = document.location;
+      presenceData.state = search.substring(9);
+      break;
+    }
+    case pathname.startsWith("/user") && pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:60]:
```suggestion
    case pathname?.startsWith("/user"): {
```

File: websites/Z/Zoro/presence.ts [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    case "/home": {
+      presenceData.details = "Exploring Zoro.to";
+      break;
+    }
+    case pages.test(pathname) && pathname: {
+      const heading: HTMLHeadElement = document.querySelector("h2.cat-heading");
+      if (heading) presenceData.details = `Looking at ${heading.innerText}`;
+      break;
+    }
+    case pathname.startsWith("/news") && pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:45]:
```suggestion
    case pathname?.startsWith("/news"): {
```

File: websites/Z/Zoro/presence.ts [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+    pages =
+      /\/(most-favorite|most-popular|movie|recently-added|recently-updated|tv|top-airing|top-upcoming|ona|ova|special|(genre\/.*))/;
+
+  switch (pathname) {
+    case "/":
+    case "/home": {
+      presenceData.details = "Exploring Zoro.to";
+      break;
+    }
+    case pages.test(pathname) && pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:40]:
```suggestion
    case pages?.test(pathname): {
```

File: websites/Z/Zoro/presence.ts [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+      const name: HTMLHeadingElement = document.querySelector("h4.name");
+      if (name) {
+        presenceData.details = `Looking at ${
+          pathname.startsWith("/people") ? "People" : "Character"
+        }`;
+        presenceData.state = name.innerText;
+      }
+      break;
+    }
+    case pathname.startsWith("/az-list") && pathname: {
```
Comment (ririxidev) [inline websites/Z/Zoro/presence.ts:80]:
```suggestion
    case pathname?.startsWith("/az-list"): {
```

**Review by rxri — CHANGES_REQUESTED** [review]

File: websites/Z/Zoro/presence.ts (hunk 2 above)
Comment (vahya) [reply websites/Z/Zoro/presence.ts:143]: >You are checking if pathname starts with "/event" so basically it is already a thing

Yes, but we need to return a string, as we are matching against `pathname` a string, `String.startsWith` returns a `boolean`.

[Check this](https://stackoverflow.com/a/43398039)

**Review by rxri — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**